app/lib/hooks/useResponsiveFontScale.tsx

```diff
@@ -0,0 +1,30 @@
+import React, { createContext, useContext } from 'react';
+import { useWindowDimensions } from 'react-native';
+
+interface IResponsiveFontScaleContextData {
+	fontScale: number;
+	isLargeFontScale: boolean;
+}
+
+interface IResponsiveFontScaleProviderProps {
+	children: React.ReactNode;
+}
+
+export const ResponsiveFontScaleContext = createContext({} as IResponsiveFontScaleContextData);
+
+const ResponsiveFontScaleProvider = ({ children }: IResponsiveFontScaleProviderProps) => {
+	const { fontScale } = useWindowDimensions();
```

Comment:
Isn't it too similar to RowHeightProvider and should be unified instead?
We could argue different contexts cause less re-renders, but since they're both tied to `useWindowDimensions' fontScale`, both are going to re-render.